Qubits in initial and final maps from `CompilationUnit` have type `UnitID` instead of `Qubit`

The conversion of some circuits is having issues. This currently fails the following tests:

```
FAILED qiskit_backend_test.py::test_state - ValueError: Requested UnitIds (request_ids) contain a mixture of qubits and bits.
FAILED qiskit_backend_test.py::test_unitary - ValueError: Requested UnitIds (request_ids) contain a mixture of qubits and bits.

--- a/pytket/binders/predicates.cpp
+++ b/pytket/binders/predicates.cpp
@@ -23,6 +23,7 @@
 #include "nanobind_json/nanobind_json.hpp"
 #include "tket/Predicates/CompilationUnit.hpp"
 #include "tket/Utils/UnitID.hpp"
+#include "typecast.hpp"
 
 namespace nb = nanobind;
 using json = nlohmann::json;

--- a/pytket/docs/changelog.rst
+++ b/pytket/docs/changelog.rst
@@ -1,6 +1,11 @@
 Changelog
 =========
 
+Unreleased
+----------
+
+Fixes: Ensure downcasting of `UnitID` to `Qubit` in initial and final maps.
+
 2.3.2 (April 2025)
 ------------------
 

--- a/pytket/tests/predicates_test.py
+++ b/pytket/tests/predicates_test.py
@@ -1191,7 +1191,6 @@ def test_zx_optimisation_wireswaps() -> None:
     assert not c.has_implicit_wireswaps
 
 
-@pytest.mark.xfail(reason="https://github.com/CQCL/tket/issues/1880")
 def test_initial_and_final_map_types() -> None:
     c = Circuit(1).H(0)
     cu = CompilationUnit(c)

